Dockerfile

```diff
@@ -7,10 +7,11 @@ RUN curl -fsSL https://www.postgresql.org/media/keys/ACCC4CF8.asc -o /etc/apt/ke
   echo "deb [signed-by=/etc/apt/keyrings/postgresql.asc] https://apt.postgresql.org/pub/repos/apt bookworm-pgdg main" >> /etc/apt/sources.list.d/pgdg.list
 RUN curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg && \
   echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_20.x nodistro main" >> /etc/apt/sources.list.d/nodesource.list
+RUN echo "Package: nodejs\nPin: origin deb.nodesource.com\nPin-Priority: 600\n\nPackage: nodejs\nPin: origin deb.debian.org\nPin-Priority: -10" > /etc/apt/preferences.d/nodejs
+RUN echo "Package: postgresql* libpq-dev\nPin: origin apt.postgresql.org\nPin-Priority: 600\n\nPackage: postgresql* libpq-dev\nPin: origin deb.debian.org\nPin-Priority: -10" > /etc/apt/preferences.d/postgresql
```

Comment:
The `-t nodistro` and `-t bookworm-pgdg` stuff I was using before seems to be anti-recommended for the most part(?), and instead you can use package pinning like this

Setting a priority less than 0 means the repo shouldn't be used. nodejs's more official install scripts set up the same pin for deb.nodesource.com, so this seems to be the recommended way to handle things, and this worked to install the packages locally (and show a problem that it couldn't find a good candidate - due to the negative pin value -when I remove the "good" repos temporarily).